proxy/enfuse/e2e_test.go

```diff
@@ -0,0 +1,113 @@
+package enfuse_test
+
+import (
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs/fstestutil"
+	"fmt"
+	"github.com/enfabrica/enkit/lib/knetwork"
+	"github.com/enfabrica/enkit/lib/srand"
+	"github.com/enfabrica/enkit/proxy/enfuse"
+	enfuse_rpc "github.com/enfabrica/enkit/proxy/enfuse/rpc"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"io/fs"
```

Comment:
```
[~/default-dev-env/bminor/dev/enkit] (fuse-share)
bminor-> bazelisk test //proxy/enfuse/...
INFO: Invocation ID: 38d02133-97ff-4576-a868-668ca53e60ea
INFO: Analyzed 5 targets (185 packages loaded, 8332 targets configured).
INFO: Found 4 targets and 1 test target...
ERROR: /home/bminor/default-dev-env/bminor/dev/enkit/proxy/enfuse/BUILD.bazel:23:8: GoCompilePkg proxy/enfuse/go_default_test_test.external.a failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -installsuffix linux_amd64 -src proxy/enfuse/e2e_test.go -arc ... (remaining 31 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
compilepkg: missing strict dependencies:
        /home/bminor/.cache/bazel/_bazel_bminor/0a7898303d30b6b2bf8d4a9ea06e14c6/sandbox/linux-sandbox/81/execroot/enkit/proxy/enfuse/e2e_test.go: import of "io/fs"
No dependencies were provided.
Check that imports in Go sources match importpath attributes in deps.
INFO: Elapsed time: 1.514s, Critical Path: 0.59s
INFO: 15 processes: 4 internal, 11 linux-sandbox.
FAILED: Build did NOT complete successfully
//proxy/enfuse:go_default_test                                  FAILED TO BUILD

FAILED: Build did NOT complete successfully
```

I bet you'll also need to bump (at least) the Go version in https://github.com/enfabrica/enkit/blob/36b0fa1525428fbb52d1df7d81efceff0c4cd8c3/bazel/init.bzl#L12

Reply:
In my dev container - I still can't run the tests. Am I doing something wrong?

```
[~/dev/enkit] (fuse-share)
sminor-> sudo apt install fuse
Reading package lists... Done
Building dependency tree
Reading state information... Done
fuse is already the newest version (2.9.9-3).
0 upgraded, 0 newly installed, 0 to remove and 179 not upgraded.
[=== 2021-10-08 22:44:19 === SUCCESS ===]
```

```
[~/dev/enkit] (fuse-share)
sminor-> bazel test //proxy/enfuse/...
INFO: Invocation ID: 96d188c7-37c3-4999-9f69-7172ed54dd0d
INFO: Analyzed 6 targets (0 packages loaded, 0 targets configured).
INFO: Found 5 targets and 1 test target...
FAIL: //proxy/enfuse:go_default_test (see /home/sminor/.cache/bazel/_bazel_sminor/9a112df0599817ddf5204210c29ce79c/execroot/enkit/bazel-out/k8-fastbuild/testlogs/proxy/enfuse/go_default_test/test.log)
INFO: From Testing //proxy/enfuse:go_default_test:
==================== Test output for //proxy/enfuse:go_default_test:
2021/10/08 22:44:25 mount helper error: fusermount: fuse device not found, try 'modprobe fuse' first
--- FAIL: TestNewFuseShareCommand (0.15s)
    e2e_test.go:47:
                Error Trace:    e2e_test.go:47
                Error:          Received unexpected error:
                                fusermount: exit status 1
                Test:           TestNewFuseShareCommand
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x6149e9]

goroutine 19 [running]:
testing.tRunner.func1.2(0x8f4ac0, 0xce3b30)
        GOROOT/src/testing/testing.go:1144 +0x332
testing.tRunner.func1(0xc000186900)
        GOROOT/src/testing/testing.go:1147 +0x4b6
panic(0x8f4ac0, 0xce3b30)
        GOROOT/src/runtime/panic.go:965 +0x1b9
bazil.org/fuse/fs/fstestutil.(*Mount).Close(0x0)
        external/org_bazil_fuse/fs/fstestutil/mounted.go:34 +0x29
panic(0x8f4ac0, 0xce3b30)
        GOROOT/src/runtime/panic.go:965 +0x1b9
proxy/enfuse/go_default_test_test.TestNewFuseShareCommand(0xc000186900)
        proxy/enfuse/e2e_test.go:51 +0x609
testing.tRunner(0xc000186900, 0x9b0590)
        GOROOT/src/testing/testing.go:1194 +0xef
created by testing.(*T).Run
        GOROOT/src/testing/testing.go:1239 +0x2b3
================================================================================
INFO: Elapsed time: 0.401s, Critical Path: 0.20s
INFO: 2 processes: 1 internal, 1 local.
INFO: Build completed, 1 test FAILED, 2 total actions
//proxy/enfuse:go_default_test                                           FAILED in 0.2s
  /home/sminor/.cache/bazel/_bazel_sminor/9a112df0599817ddf5204210c29ce79c/execroot/enkit/bazel-out/k8-fastbuild/testlogs/proxy/enfuse/go_default_test/test.log

INFO: Build completed, 1 test FAILED, 2 total actions
[=== 2021-10-08 22:44:25 === FAIL: 3 ===]
```

```
[~/dev/enkit] (fuse-share)
sminor-> modprobe fuse
modprobe: FATAL: Module fuse not found in directory /lib/modules/5.4.131-16049-gc688e114d34e
[=== 2021-10-08 22:44:41 === FAIL: 1 ===]
```

Reply:
IIRC you need to `apt update` and reboot the container after installing FUSE. 

Assuming you did that, can you post the permissions of fusermount?